config/load.go

```diff
@@ -11,87 +11,68 @@ import (
 	"github.com/spf13/viper"
 )
 
-// isValidEnvironment will return true if the testEnv is a known valid environment
-func isValidEnvironment(testEnv string) bool {
-	testEnv = strings.ToLower(testEnv)
-	for _, env := range environments {
-		if env == testEnv {
-			return true
-		}
-	}
-	return false
-}
-
-// getWorkingDirectory will get the current working directory
-func getWorkingDirectory() string {
-	dir, err := os.Getwd()
-	if err != nil {
-		return ""
-	}
-	return dir
-}
-
 // Added a mutex lock for a race-condition
 var viperLock sync.Mutex
 
-// Load all environment variables
-func Load(customWorkingDirectory string) (_appConfig *AppConfig, err error) {
+// Load all AppConfig
+func Load(configFilePath string) (appConfig *AppConfig, err error) {
+	viperLock.Lock()
+	defer viperLock.Unlock()
 
-	// Check the environment we are running
-	environment := os.Getenv(EnvironmentKey)
-	if !isValidEnvironment(environment) {
-		err = fmt.Errorf(dictionary.GetInternalMessage(dictionary.ErrorInvalidEnv), environment)
-		logger.Data(2, logger.ERROR, err.Error())
-		return
-	}
+	setDefaults(configFilePath)
 
-	// Get the working directory
-	var workingDirectory string
-	if len(customWorkingDirectory) > 0 {
-		workingDirectory = customWorkingDirectory
-	} else {
-		workingDirectory = getWorkingDirectory()
+	if err = loadFlags(); err != nil {
```

Comment:
As we're setting defaults on a flags definitions, do we still need this setDefault function?

Reply:
In theory that would work.

But in practice, I think that it's not the flags responsibility to set global defaults, because flags are not directly from viper, we just bind them. Viper defaults should be the source of truth, not flag defaults. We may have default values that we don't want to expose as flags.

Using the same defaults in flags and in Viper settings is just a convention I used that I think is the most elegant of all options.